environments/common/src/main/java/org/jboss/weld/environment/deployment/discovery/ClassPathBeanArchiveScanner.java

```diff
@@ -114,11 +126,20 @@ private void scanDirectory(File entryDirectory, ImmutableList.Builder<ScanResult
                 results.add(new ScanResult(null, entryDirectory.getPath()));
             }
         }
+
+        File manifestFile = new File(entryDirectory, MANIFEST_FILE);
+        if (manifestFile.canRead()) {
+            final Manifest manifest = new Manifest(new FileInputStream(manifestFile));
```

Comment:
You are never closing the `FileInputStream`.

Reply:
Oups, my bad, hacked this together too fast.
Good catch.
Already pushed a fix.

Reply:
Thanks!